### PR TITLE
Feature/web log analytics app

### PR DESCRIPTION
**Pull request by gsps1**
Moved Doc to README, updated flowlet name to be consistent with the guides standard

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+You will build a CDAP application that uses web logs aggregated by Flume to find page view counts. You will:
+
+* Configure Flume to ingest data into `CDAP Stream <http://docs.cask.co/cdap/current/en/dev-guide.html#streams>`__
+* Build real-time `Flow <http://docs.cask.co/cdap/current/en/dev-guide.html#flows>`__ to process ingested web logs
+* Build a `Service <http://docs.cask.co/cdap/current/en/dev-guide.html#services>`__ to serve analysis results via HTTP
+
+What you will need
+------------------
+* `JDK 6 or JDK 7 <http://www.oracle.com/technetwork/java/javase/downloads/index.html>`__
+* `Apache Maven 3.0+ <http://www.google.com/url?q=http%3A%2F%2Fmaven.apache.org%2F&sa=D&sntz=1&usg=AFQjCNGhqqn4qI1Qjh2sLRjeezXuA2e20Q>`__
```
Comment (abaranau) [inline README.rst]: no google - link to maven directly

File: README.rst [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+*  CDAP SDK
+*  `Apache Flume <http://www.google.com/url?q=http%3A%2F%2Fflume.apache.org%2Fdownload.html&sa=D&sntz=1&usg=AFQjCNEvincOaiFtXXxkdGgo3cOUqrsJtg>`__
```
Comment (abaranau) [inline README.rst]: same

File: README.rst [hunk 3]
```diff
@@ -0,0 +1,239 @@
...
+
+Let’s Build It!
+---------------
+The following sections will guide you through configuring and running Flume , and  implementing an application from scratch. 
+If you want to deploy and run application right away, you can download full sources and binaries of the application at <link>.
```
Comment (abaranau) [inline README.rst]: "you can clone sources from this github repository".

File: README.rst [hunk 4]
```diff
@@ -0,0 +1,239 @@
...
+
+Let’s Build It!
+---------------
+The following sections will guide you through configuring and running Flume , and  implementing an application from scratch. 
+If you want to deploy and run application right away, you can download full sources and binaries of the application at <link>. 
+In this case feel free to skip following two sections and jump to Build and Run Application section
+
+Application Design
+------------------
+Web logs are aggregated using Flume which pushes the data to a webLogs Stream using special StreamSink from `cdap-ingest <https://github.com/caskdata/cdap-ingest>`__ library.
```
Comment (abaranau) [inline README.rst]: do we host javadocs of cdap-ingest anywhere? so that you can link StreamSink to them here

File: README.rst [hunk 5]
```diff
@@ -0,0 +1,239 @@
...
+Let’s Build It!
+---------------
+The following sections will guide you through configuring and running Flume , and  implementing an application from scratch. 
+If you want to deploy and run application right away, you can download full sources and binaries of the application at <link>. 
+In this case feel free to skip following two sections and jump to Build and Run Application section
+
+Application Design
+------------------
+Web logs are aggregated using Flume which pushes the data to a webLogs Stream using special StreamSink from `cdap-ingest <https://github.com/caskdata/cdap-ingest>`__ library. 
+Then, logs are processed in real-time with a Flow that consumes data from webLogs stream and persists computation results in a pageViews `Dataset <http://docs.cask.co/cdap/current/en/dev-guide.html#datasets>`__.
```
Comment (abaranau) [inline README.rst]: either don't link to dataset or add link to flow word as well...

File: README.rst [hunk 6]
```diff
@@ -0,0 +1,239 @@
...
+    <app_dir>/src/main/java/co/cask/cdap/guide/WebLogAnalyticsHandler.java
+
+WebLogAnalyticsApplication declares that the application has a stream, flow, service and uses a dataset::
+
+  public class WebLogAnalyticsApplication extends AbstractApplication {
+  
+    @Override
+    public void configure() {
+      setName("WebLogAnalyticsApp");
+      setDescription("Application to perform web log analytics using the Cask Data Application Platform");
```
Comment (abaranau) [inline README.rst]: remove - description doesn't add value, but make code look huge

File: README.rst [hunk 7]
```diff
@@ -0,0 +1,239 @@
...
+      addStream(new Stream("webLogStream"));
```
Comment (abaranau) [inline README.rst]: "webLogs"

File: README.rst [hunk 8]
```diff
@@ -0,0 +1,239 @@
...
+The recommended way to build a CDAP application from scratch is to use maven project. Use the following directory structure::
+  
+    <app_dir>/pom.xml
+    <app_dir>/src/main/java/co/cask/cdap/guide/WebLogAnalyticsApplication.java
+    <app_dir>/src/main/java/co/cask/cdap/guide/WebLogAnalyticsFlow.java
+    <app_dir>/src/main/java/co/cask/cdap/guide/WebLogAnalyticsFlowlet.java
+    <app_dir>/src/main/java/co/cask/cdap/guide/WebLogAnalyticsHandler.java
+
+WebLogAnalyticsApplication declares that the application has a stream, flow, service and uses a dataset::
+
```
Comment (abaranau) [inline README.rst:51]: everywhere: tell rst to use Java syntax highlighting - see timeseries guide README

File: README.rst [hunk 9]
```diff
@@ -0,0 +1,239 @@
...
+  
+WebLogAnalyticsFlow makes use of  WebLogAnalyticsFlowlet::
+
+  public class WebLogAnalyticsFlow implements Flow {
+    @Override
+    public FlowSpecification configure() {
+       return FlowSpecification.Builder.with().
+        setName("WebLogAnalyticsFlow").
+        setDescription("A flow that collects and processes weblogs").
+        withFlowlets().add("WebLogAnalyticsFlowlet", new
```
Comment (abaranau) [inline README.rst]: use "analyze" as flowlet name please

Comment (abaranau) [inline README.rst]: Or better "pageViewCounter" as flowlet name. And PageViewCounterFlowlet as class name.

File: README.rst [hunk 10]
```diff
@@ -0,0 +1,239 @@
...
+      Matcher requestMatcher = REQUEST_PAGE_PATTERN.matcher(request);
+      if (!requestMatcher.matches() || requestMatcher.groupCount() < 2) {
+        LOG.info("Invalid event received {}", log);
+        return;
+      }
+      String uri = requestMatcher.group(2);
+      pageViewTable.increment(Bytes.toBytes(uri), 1L);
+    }
+  }
+For example, given the following event::
```
Comment (abaranau) [inline README.rst:116]: empty line missing

File: README.rst [hunk 11]
```diff
@@ -0,0 +1,239 @@
...
+      }
+      responder.sendString(200, pageViews.toString(), Charsets.UTF_8);
+    }
+  }
+
+Build and Run Application
+-------------------------
+Now that your application implementation is ready, you can build and deploy to CDAP::
+
+  > mvn clean package
```
Comment (abaranau) [inline README.rst]: remove ">" - make commands copy-paste-able

File: README.rst [hunk 12]
```diff
@@ -0,0 +1,239 @@
...
+       KeyValue<byte[], byte[]> uri = pageViewScan.next();
+       pageViews.put(new String(uri.getKey()), Bytes.toLong(uri.getValue()));
+      }
+      responder.sendString(200, pageViews.toString(), Charsets.UTF_8);
+    }
+  }
+
+Build and Run Application
+-------------------------
+Now that your application implementation is ready, you can build and deploy to CDAP::
```
Comment (abaranau) [inline README.rst]: build it and...

File: README.rst [hunk 13]
```diff
@@ -0,0 +1,239 @@
...
+
+  > mvn clean package
+  > bin/cdap-cli.sh deploy app WebLogAnalyticsApplication.jar
```
Comment (abaranau) [inline README.rst]: can you make consistent with timeseries guide - very minor - there's EXPORT, etc.

File: README.rst [hunk 14]
```diff
@@ -0,0 +1,239 @@
...
+  > bin/cdap-cli.sh start service WebLogAnalyticsApp.WebLogAnalyticsService
+
+Once the flow is started, it is ready to receive the web logs from stream. Now let’s configure and start Flume to push web logs into a Stream.
+
+Ingest Data with Flume
+----------------------
+In the provided sources for this guide you can find Apache web server’s access.log file that we will use as a source of data. If you have access to live Apache web server’s access logs you can use them instead.
+
+In order to configure Apache Flume to push web logs to a CDAP Stream you need to create a simple flow which includes:
+source that tail access logs
```
Comment (abaranau) [inline README.rst]: shouldn't it be a list?

Comment (abaranau) [inline README.rst]: I mean bulletted list formatting

File: README.rst [hunk 15]
```diff
@@ -0,0 +1,239 @@
...
+--------------------
+Download CDAP flume sink jar::
+
+  cd <flume-base-dir>/lib
+  curl --remote-name https://oss.sonatype.org/content/repositories/releases/co/cask/cdap/cdap-flume/1.0.1/cdap-flume-1.0.1.jar
+
+CDAP Flume sink requires newer version of `Guava <https://code.google.com/p/guava-libraries/>`__ library than that is usually shipped with Flume. You need to replace the existing guava library with guava-17.0.jar::
+
+  mkdir <flume-base-dir>/backup-lib
+  mv <flume-base-dir>/lib/guava-<existing-version>.jar  <flume-base-dir>/backup-lib
```
Comment (abaranau) [inline README.rst]: just remove

Comment (abaranau) [inline README.rst]: I mean no mv, rm -r

**Comment by abaranau** [comment]
couple comments

File: README.rst (hunk 4 above)
Comment (gsps1) [inline README.rst]: its not there yet

File: README.rst (hunk 5 above)
Comment (gsps1) [inline README.rst]: flow is linked the first time its referenced, line #12

Comment (gsps1) [inline README.rst]: flow is already linked at #12

Comment (abaranau) [inline README.rst]: but dataset too, no?

Comment (gsps1) [inline README.rst]: not linked, but prepackaged Datasets is mentioned in first paragraph, will link dataset to that instead.

**Comment by gsps1** [comment]
addressed the comments. merging.

**Pull request merged**